src/liiteri/mime.clj

```diff
@@ -35,11 +51,12 @@
        (str "Could not get extension for content-type '" real-content-type-name "', exception: " (.getMessage e)))
       (first (fs/split-ext filename)))))
 
-(defn file->validated-file-spec! [config filename tempfile size]
-  (let [detected-content-type (detect-mime-type tempfile)
-        updated-filename      (fix-extension filename detected-content-type)]
-    (validate-file-content-type! config updated-filename detected-content-type)
-    {:content-type detected-content-type
-     :filename     updated-filename
-     :size         size
-     :tempfile     tempfile}))
+(defn file->validated-file-spec! [config filename get-file]
+  (let [{:keys [size file]} (get-file)
+        detected-content-type (detect-mime-type #(identity file))
```

Comment:
Ja sit täällä 
```suggestion
        detected-content-type (detect-mime-type file)
```

Reply:
Mut jos jostain syystä halutaan myöhäistää tuota get-streamin kutsumista, niin sitten musta parempi näin:
```suggestion
        detected-content-type (detect-mime-type (fn [] file))
```